Build failure: vaporizer2

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

I just have the package in my config, and it failed to build on upgrade.

### Can Hydra reproduce this build failure?

The last build job was manually cancelled.

### Link to Hydra build job

https://hydra.nixos.org/build/315672159

### Relevant log output

```console
vaporizer2> [ 97%] Building CXX object CMakeFiles/VASTvaporizer2_Standalone.dir/nix/store/7j4793x4la0qv32xrjwl9p6xd7zixzs8-juce-8.0.11/share/juce/modules/juce_audio_plugin_client/juce_audio_plugin_client_VST2.cpp.o
vaporizer2> [ 98%] Building CXX object CMakeFiles/VASTvaporizer2_Standalone.dir/nix/store/7j4793x4la0qv32xrjwl9p6xd7zixzs8-juce-8.0.11/share/juce/modules/juce_audio_plugin_client/juce_audio_plugin_client_VST3.cpp.o
vaporizer2> [ 98%] Building CXX object CMakeFiles/VASTvaporizer2_VST3.dir/nix/store/7j4793x4la0qv32xrjwl9p6xd7zixzs8-juce-8.0.11/share/juce/modules/juce_audio_plugin_client/juce_audio_plugin_client_VST3.cpp.o
vaporizer2> /build/source/clap-juce-extensions/src/wrapper/clap-juce-wrapper.cpp:21:10: fatal error: juce_audio_processors/format_types/juce_LegacyAudioParameter.cpp: No such file or directory
vaporizer2>    21 | #include <juce_audio_processors/format_types/juce_LegacyAudioParameter.cpp>
vaporizer2>       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
vaporizer2> compilation terminated.
vaporizer2> make[2]: *** [CMakeFiles/VASTvaporizer2_CLAP.dir/build.make:79: CMakeFiles/VASTvaporizer2_CLAP.dir/clap-juce-extensions/src/wrapper/clap-juce-wrapper.cpp.o] Error 1
vaporizer2> make[1]: *** [CMakeFiles/Makefile2:408: CMakeFiles/VASTvaporizer2_CLAP.dir/all] Error 2
vaporizer2> make[1]: *** Waiting for unfinished jobs....
vaporizer2> [ 98%] Linking CXX executable VASTvaporizer2_artefacts/Release/Standalone/VASTvaporizer2
vaporizer2> [ 98%] Built target VASTvaporizer2_Standalone
vaporizer2> [ 98%] Linking CXX shared module VASTvaporizer2_artefacts/Release/LV2/VASTvaporizer2.lv2/libVASTvaporizer2.so
vaporizer2> Fontconfig error: Cannot load default config file: No such file: (null)
vaporizer2>
vaporizer2> Fontconfig error: No writable cache directories
vaporizer2>         /var/cache/fontconfig
vaporizer2>         /homeless-shelter/.cache/fontconfig
vaporizer2>
vaporizer2> Fontconfig error: No writable cache directories
vaporizer2>         /var/cache/fontconfig
vaporizer2>         /homeless-shelter/.cache/fontconfig
vaporizer2>
vaporizer2> Fontconfig error: No writable cache directories
vaporizer2>         /var/cache/fontconfig
vaporizer2>         /homeless-shelter/.cache/fontconfig
vaporizer2>
vaporizer2> Fontconfig error: No writable cache directories
vaporizer2>         /var/cache/fontconfig
vaporizer2>         /homeless-shelter/.cache/fontconfig
vaporizer2>
vaporizer2> Fontconfig error: No writable cache directories
vaporizer2>         /var/cache/fontconfig
vaporizer2>         /homeless-shelter/.cache/fontconfig
vaporizer2>
vaporizer2> Fontconfig error: No writable cache directories
vaporizer2>         /var/cache/fontconfig
vaporizer2>         /homeless-shelter/.cache/fontconfig
vaporizer2>
vaporizer2> Fontconfig error: No writable cache directories
vaporizer2>         /var/cache/fontconfig
vaporizer2>         /homeless-shelter/.cache/fontconfig
vaporizer2>
vaporizer2> Fontconfig error: No writable cache directories
vaporizer2>         /var/cache/fontconfig
vaporizer2>         /homeless-shelter/.cache/fontconfig
vaporizer2> open failure: No such file or directory
vaporizer2> [ 98%] Built target VASTvaporizer2_LV2
vaporizer2> [ 99%] Linking CXX shared module VASTvaporizer2_artefacts/Release/VST3/VASTvaporizer2.vst3/Contents/x86_64-linux/VASTvaporizer2.so
vaporizer2> removing moduleinfo.json
vaporizer2> creating /build/source/build/VASTvaporizer2_artefacts/Release/VST3/VASTvaporizer2.vst3
vaporizer2> [ 99%] Built target VASTvaporizer2_VST3
vaporizer2> make: *** [Makefile:136: all] Error 2
```

### Additional context

There where both canceled and failed builds, so I linked to the failed build, hope that helps.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.60, NixOS, 26.05 (Yarara), 26.05.20251202.418468a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"nixos, nixos-hardware"`
 - nixpkgs: `/nix/store/xsdf4r7hl94n1a1kjj02ibs611jnmadn-source`


### Notify maintainers

@eljamm 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ai/airwin2rack/package.nix
+++ b/pkgs/by-name/ai/airwin2rack/package.nix
@@ -49,8 +49,8 @@ let
   clapJuceExtensions = fetchFromGitHub {
     owner = "free-audio";
     repo = "clap-juce-extensions";
-    rev = "4f33b4930b6af806018c009f0f24b3a50808af99";
-    hash = "sha256-M+T7ll3Ap6VIP5ub+kfEKwT2RW2IxxY4wUPRQKFIotk=";
+    rev = "645ed2fd0949d36639e3d63333f26136df6df769";
+    hash = "sha256-Lx88nyEFjPLA5yh8rrqBdyZIxe/j0FgIHoyKcbjuuI4=";
     fetchSubmodules = true;
   };
 

--- a/pkgs/by-name/va/vaporizer2/package.nix
+++ b/pkgs/by-name/va/vaporizer2/package.nix
@@ -30,6 +30,15 @@ stdenv.mkDerivation (finalAttrs: {
     fetchSubmodules = true;
   };
 
+  postPatch = ''
+    # LTO needs special setup on Linux
+    substituteInPlace CMakeLists.txt \
+      --replace-fail 'juce::juce_recommended_lto_flags' '# Not forcing LTO'
+
+    rm -rf clap-juce-extensions
+    ln -s ${finalAttrs.passthru.clapJuceExtensions} clap-juce-extensions
+  '';
+
   strictDeps = true;
 
   nativeBuildInputs = [
@@ -56,11 +65,13 @@ stdenv.mkDerivation (finalAttrs: {
     (lib.cmakeFeature "USE_SYSTEM_JUCE" "ON")
   ];
 
-  # LTO needs special setup on Linux
-  postPatch = ''
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'juce::juce_recommended_lto_flags' '# Not forcing LTO'
-  '';
+  passthru.clapJuceExtensions = fetchFromGitHub {
+    owner = "free-audio";
+    repo = "clap-juce-extensions";
+    rev = "645ed2fd0949d36639e3d63333f26136df6df769";
+    hash = "sha256-Lx88nyEFjPLA5yh8rrqBdyZIxe/j0FgIHoyKcbjuuI4=";
+    fetchSubmodules = true;
+  };
 
   meta = {
     description = "Wavetable synthesizer";

